packages/codec/lib/core.ts

```diff
@@ -51,7 +51,8 @@ export function* decodeVariable(
  */
 export function* decodeCalldata(
   info: Evm.EvmInfo,
-  isConstructor?: boolean //ignored if context! trust context instead if have
+  isConstructor?: boolean, //ignored if context! trust context instead if have
+  strictAbiMode?: boolean //used for selector-based decoding
```

Comment:
I think we need to document somewhere what this option does, since it kind of turns this function into something of a chimera. Like, how is it used for selector-based decoding?

Reply:
Hm.  I mean all of this stuff doesn't really have documentation, as this is codec rather than decoder.  But I can expand the comment to explain it better.